llvm/include/llvm/CodeGen/TargetLowering.h

```diff
@@ -2273,6 +2275,18 @@ class LLVM_ABI TargetLoweringBase {
         "Generic atomicrmw expansion unimplemented on this target");
   }
 
+  /// Perform a atomic store using a target-specific way.
+  virtual void emitExpandAtomicStore(StoreInst *SI) const {
```

Comment:
These atomic APIs are a huge mess. Do we really need the same API for every single type of atomic instruction? I guess this just adds another, merging them should be a separate change